syntax = "proto3";

package monitoring;

service MonitorService {
    rpc StreamMetrics (stream MetricMessage) returns (stream CommandMessage);
}

message MetricMessage {
    string timestamp = 1;
    string hostname = 2;
    string metric = 3;
    float value = 4;
}

// Cấu trúc lệnh từ Server gửi về (ví dụ đơn giản)
message CommandMessage {
    string command_id = 1;
    string action = 2;
}